TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php

```diff
@@ -461,17 +461,17 @@ protected function replaceNodeData(NodeInterface $node, NodeData $targetNodeData
             }
         }
 
-        if ($node->isRemoved() === true) {
-            $this->nodeDataRepository->remove($targetNodeData);
-        } else {
-            $targetNodeData->similarize($node->getNodeData());
-            if ($nodeWasMoved) {
-                $targetNodeData->setPath($node->getPath(), false);
-            }
-            $targetNodeData->setLastPublicationDateTime($this->now);
-            $node->setNodeData($targetNodeData);
-            $this->nodeService->cleanUpProperties($node);
+        $targetNodeData->setRemoved($node->isRemoved());
+        $targetNodeData->similarize($sourceNodeData);
+        // TODO: This seemswrong and introduces a publish order between nodes. We should always set the path.
+        if ($nodeWasMoved) {
+            $targetNodeData->setPath($node->getPath(), false);
         }
+        $targetNodeData->setLastPublicationDateTime($this->now);
+        $node->setNodeData($targetNodeData);
+        $node->setNodeDataIsMatchingContext(null);
```

Comment:
That method doesn't exist in `NodeInterface` or am I missing a change? And what is that line supposed to do?